argus/recipees/cloud/windows.py

```diff
@@ -64,19 +64,34 @@ class WindowsCloudbaseinitRecipee(base.BaseCloudbaseinitRecipee):
     """Recipee for preparing a Windows instance."""
 
     def get_program_files(self):
-        """Get the location of program files from the instance."""
+        """Get the location of program files from the instance.
+
+        Warning: it's recommended to call this after cb-init is installed.
+        """
         stdout, _ = self._execute(
             'powershell "(Get-WmiObject  Win32_OperatingSystem).'
             'OSArchitecture"')
         architecture = stdout.strip()
 
         # Next, get the location.
+        locations = [self._execute('powershell "$ENV:ProgramFiles"')[0]]
         if architecture == '64-bit':
             location, _ = self._execute(
                 'powershell "${ENV:ProgramFiles(x86)}"')
-        else:
-            location, _ = self._execute('powershell "$ENV:ProgramFiles"')
-        return location.strip()
+            locations.append(location)
+
+        _locations = []
+        for location in locations:
+            location = location.strip()
+            _location = location.replace(" ", "` ")
+            status = self._execute(
+                'powershell Test-Path "{}\\Cloudbase` Solutions"'.format(
+                    _location)).strip().lower()
+            if status == "true":
+                _locations.append(location)
+        _locations = _locations or locations
+
+        return _locations[-1].strip()
```

Comment:
This is overcomplicated. Iterate through each location, see if it exists and if it exists, break out of loop.